### PR TITLE
Support for setting task Date/Time variables

### PR DESCRIPTION
**Pull request by CTI777**
Related to https://github.com/Activiti/Activiti/issues/2876

### CODE REVIEW

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/TaskVariablesPayloadValidator.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    
+    public void checkPayloadVariables(Map<String, Object> variablePayloadMap) {
+       if (variablePayloadMap != null) {
+           for (Map.Entry<String, Object> payloadVar : variablePayloadMap.entrySet()) {
+                Object value = payloadVar.getValue();
+                           
+                if (value instanceof String) {
+                    try {
+                        payloadVar.setValue(dateFormatterProvider.parse((String)value));
+                    } catch (Exception e) {}
```
Comment (ffazzini) [inline activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/TaskVariablesPayloadValidator.java]: i see a lot of empty catch blocks in this class, should not we log something or re-throw an handled exception?

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/TaskVariablesIT.java [hunk 2]
```diff
@@ -195,20 +200,94 @@ public void adminShouldSetGetUpdateTaskVariables() {
         assertThat(variablesContainEntry("var1","test1",variablesResponse.getBody().getContent())).isTrue();
         assertThat(variablesContainEntry("var3","test3-update",variablesResponse.getBody().getContent())).isTrue();
     }
+    
+    @Test
+    public void should_Create_Update_DateTimeTaskVariables() throws Exception{
```
Comment (ffazzini) [inline activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/TaskVariablesIT.java]: minor one, can we stick to the naming "should_..._When_...." ?

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/TaskVariablesPayloadValidator.java (hunk 1 above)
Comment (CTI777) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/TaskVariablesPayloadValidator.java]: I agree that we have to handle exceptions proper... But in these cases we try to convert String to Date, and if it fails, we simply keep variable as a string

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/TaskVariablesIT.java (hunk 2 above)
Comment (CTI777) [reply activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/TaskVariablesIT.java]: Have try to use other names (don't know if they are better)

File: activiti-cloud-starter-runtime-bundle/src/test/resources/processes/task-date-var-mapping-extensions.json [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        "type": "date",
+        "required": true,
+        "value": "2019-09-01"
+      },
+      "process-variable-datetime-id": {
+        "id": "process-variable-datetime-id",
+        "name": "process-variable-datetime",
+        "type": "date",
+        "required": true,
+        "value": "2019-09-01"
```
Comment (erdemedeiros) [inline activiti-cloud-starter-runtime-bundle/src/test/resources/processes/task-date-var-mapping-extensions.json]: Now you can specify a date with time here, right?

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
Just a question.

File: activiti-cloud-starter-runtime-bundle/src/test/resources/processes/task-date-var-mapping-extensions.json (hunk 3 above)
Comment (CTI777) [reply activiti-cloud-starter-runtime-bundle/src/test/resources/processes/task-date-var-mapping-extensions.json]: Yes we can... After corresponding PR (activiti-core) was merged, it is possible... Should I apply this here? This check was made in other PR...

Comment (erdemedeiros) [reply activiti-cloud-starter-runtime-bundle/src/test/resources/processes/task-date-var-mapping-extensions.json]: Please, apply it here as well.. No need to to check the actual value, just change the format (for consistency). You can use zeros to keep the same date if you prefer.

Comment (CTI777) [reply activiti-cloud-starter-runtime-bundle/src/test/resources/processes/task-date-var-mapping-extensions.json]: Changed

**Review by erdemedeiros — APPROVED** [review]

**Review by ffazzini — APPROVED** [review]

**Pull request merged**